lib/components/custom_switch.dart

```diff
@@ -0,0 +1,80 @@
+import 'package:flutter/material.dart';
+
+enum Option { first, second }
+
+class CustomSwitch extends StatefulWidget {
+  const CustomSwitch(
+      {super.key,
+      required this.firstScreen,
+      required this.secondScreen,
+      required this.firstLabel,
+      required this.secondLabel});
+
+  final Widget firstScreen;
+  final Widget secondScreen;
+  final String firstLabel;
+  final String secondLabel;
+
+  @override
+  _CustomSwitchState createState() => _CustomSwitchState();
+}
+
+class _CustomSwitchState extends State<CustomSwitch> {
+  bool isSwitched = false;
```

Comment:
@IgorJonski  czy my jakoś w takim wypadku mamy korzystać z Riverpoda?

Reply:
nieee, jak to jest jakiś state wewnątrz tylko jednego widgetu i inne elementy apki nie są od niego zależne to używamy zwykłego state z StatefulWidget.